AG-16239 - BarSeries performance improvements pt. 2

alantreadway opened the pull request:
https://ag-grid.atlassian.net/browse/AG-16239

alantreadway committed:
AG-16239 - Optimise DomainManager.processValue().

alantreadway committed:
AG-16239 - Optimise DomainManager.processValue().

alantreadway committed:
AG-16239 - Add reducer + processor incremental support.

alantreadway committed:
AG-16239 - Add reducer + processor incremental support.

alantreadway committed:
AG-16239 - Add reducer + processor incremental support.

alantreadway committed:
AG-16239 - Add reducer + processor incremental support.

alantreadway committed:
AG-16239 - Add reducer + processor incremental support.

alantreadway committed:
AG-16239 - Add reducer + processor incremental support.

alantreadway committed:
AG-16239 - Add reducer + processor incremental support.

alantreadway committed:
Delete packages/ag-charts-community/src/chart/data/data-model/reducers/DESIGN_banded_interval_reducers.md

alantreadway committed:
AG-16239 - Add reducer + processor incremental support.

alantreadway committed:
AG-16239 - Add reducer + processor incremental support.

alantreadway committed:
AG-16239 - Add reducer + processor incremental support.

alantreadway committed:
AG-16239 - Add reducer + processor incremental support.

alantreadway committed:
AG-16239 - Add reducer + processor incremental support.

alantreadway committed:
AG-16239 - Add reducer + processor incremental support.

alantreadway committed:
AG-16239 - Add reducer + processor incremental support.

alantreadway committed:
AG-16239 - Add reducer + processor incremental support.

alantreadway committed:
AG-16239 - Add reducer + processor incremental support.

lsjroberts commented on packages/ag-charts-community/src/chart/data/data-model/domain/domainManager.ts:
```diff
@@ -1,11 +1,15 @@
+import { ProcessValueFactory } from './processValueFactory';
+import type { SpecializedProcessValueFn } from './processValueFactory';
+
+export type { SpecializedProcessValueFn } from './processValueFactory';
```
Why re-export this?

alantreadway replied:
Good spot, fixed.

lsjroberts commented on packages/ag-charts-community/src/chart/data/data-model/domain/processValueFactory.ts:
```diff
@@ -0,0 +1,548 @@
+     * Creates a specialized processValue function optimized for key properties with validation.
+     * Eliminates all branching for the most common key property case (~30% of calls).
+     */
+    private createSpecializedProcessValue_Key_Validation(
```
why the `_`? (and elsewhere)

alantreadway replied:
Changed to camelCase - thanks!

lsjroberts reviewed (approved):
LGTM

alantreadway committed:
Address PR feedback: remove type re-export and convert function names to camelCase

- Remove unnecessary type re-export of SpecializedProcessValueFn from domainManager
- Update imports in dataExtractor and incrementalProcessor to import directly from processValueFactory
- Convert all function names from snake_case to camelCase (e.g., processValue_Key_Validation_Accessor -> processValueKeyValidationAccessor)

alantreadway commented:
@codex PTAL
